python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -341,43 +426,60 @@ def applyAmcvn(self, params, expmjd_in):
+        burst_freq = params['burst_freq'].astype(float)[valid_dexes]
+        burst_scale = params['burst_scale'].astype(float)[valid_dexes]
+        amp_burst = params['amp_burst'].astype(float)[valid_dexes]
+        color_excess = params['color_excess_during_burst'].astype(float)[valid_dexes]
+        does_burst = params['does_burst'][valid_dexes]
+
         # get the light curve of the typical variability
-        uLc   = params['amplitude']*numpy.cos((epochs - params['t0'])/params['period'])
-        gLc   = uLc
-        rLc   = uLc
-        iLc   = uLc
-        zLc   = uLc
-        yLc   = uLc
+        uLc   = amplitude*numpy.cos((epoch - t0)/period)
+        gLc   = copy.deepcopy(uLc)
+        rLc   = copy.deepcopy(uLc)
+        iLc   = copy.deepcopy(uLc)
+        zLc   = copy.deepcopy(uLc)
+        yLc   = copy.deepcopy(uLc)
 
         # add in the flux from any bursting
-        if params['does_burst']:
-            adds = numpy.zeros(epochs.size)
-            for o in numpy.linspace(params['t0'] + params['burst_freq'],\
-                                 params['t0'] + maxyears*365.25, \
-                                 numpy.ceil(maxyears*365.25/params['burst_freq'])):
-                tmp = numpy.exp( -1*(epochs - o)/params['burst_scale'])/numpy.exp(-1.)
-                adds -= params['amp_burst']*tmp*(tmp < 1.0)  ## kill the contribution
+        local_bursting_dexes = numpy.where(does_burst==1)
+        for i_burst in local_bursting_dexes[0]:
+            adds = 0.0
+            for o in numpy.linspace(t0[i_burst] + burst_freq[i_burst],\
+                                 t0[i_burst] + maxyears*365.25, \
+                                 numpy.ceil(maxyears*365.25/burst_freq[i_burst])):
+                tmp = numpy.exp( -1*(epoch - o)/burst_scale[i_burst])/numpy.exp(-1.)
+                adds -= amp_burst[i_burst]*tmp*(tmp < 1.0)  ## kill the contribution
             ## add some blue excess during the outburst
-            uLc += adds +  2.0*params['color_excess_during_burst']*adds/min(adds)
-            gLc += adds + params['color_excess_during_burst']*adds/min(adds)
-            rLc += adds + 0.5*params['color_excess_during_burst']*adds/min(adds)
-            iLc += adds
-            zLc += adds
-            yLc += adds
-
-        dMag['u'] = uLc
-        dMag['g'] = gLc
-        dMag['r'] = rLc
-        dMag['i'] = iLc
-        dMag['z'] = zLc
-        dMag['y'] = yLc
+            uLc[i_burst] += adds +  2.0*color_excess[i_burst]
+            gLc[i_burst] += adds + color_excess[i_burst]
+            rLc[i_burst] += adds + 0.5*color_excess[i_burst]
+            iLc[i_burst] += adds
+            zLc[i_burst] += adds
+            yLc[i_burst] += adds
+
+        dMag[0][valid_dexes] += uLc
+        dMag[1][valid_dexes] += gLc
+        dMag[2][valid_dexes] += rLc
+        dMag[3][valid_dexes] += iLc
+        dMag[4][valid_dexes] += zLc
+        dMag[5][valid_dexes] += yLc
         return dMag
```

Comment:
I have run all of the example notebooks we generally give to people.  Everything is fine.  I don't think we ever told the general user base that we were using dicts, so we shouldn't need to worry about people writing their own 'applyVariability' methods that pass out dicts instead of arrays.

Also: our unit test coverage is pretty good, so I think we are fine.